### ISSUE TITLE
Conda tab completion

### ISSUE THREAD

**Issue by adrienpacifico** [conda/conda-docs#635]
[Conda doc](https://conda.io/docs/user-guide/configuration/enable-tab-completion.html) state that tab completion is deprecated since Conda 4.4 .

Does this mean that Conda is not supposed to work with tab completion ?

(Fresh install on brand new Mac, tab completion is not working) 

Why is tab completion deprecated ? I tend to expect bash command line programs to always have so sort of tab completion. Can someone explain ?

Best

**Comment by 1oglop1** [comment]
Yes tab completion is quite vital functionality, probably nobody likes to type long commands and mess up with 1 stupid typo.

there are few more notes in #415 
Tab completion should be there!

**Comment by kalefranz** [comment]
Tab completion is available for fish shell. Writing bash completions would be a good first project for someone, referencing how fish implemented theirs. Pull requests welcome.

**Comment by noahp** [comment]
Fish completions are here it looks like FWIW:
https://github.com/conda/conda/blob/master/conda/shell/etc/fish/conf.d/conda.fish#L83-L139

**Comment by tartansandal** [comment]
Implemented basic bash completion based on the completion for fish PR at https://github.com/conda/conda/pull/9139
